### ISSUE TITLE
[PROPOSAL] Support DOM properties binding as HTML attribute.

### ISSUE THREAD

**Issue by montyclt** [aspnet/Blazor#649]
A DOM object has attributes (set by HTML attributes) and properties, set or get by dot notation in DOM object.

An example:

```
// HTML
<input id="name" value="Monty" />

// Assume I render the page and change the content of textbox from Monty by Anna.

// JavaScript
var name = document.getElementById("name");

name.getAttribute("value"); // Monty
name.value; // Anna
```

As see, the value of the attribute is not the same than the property.

Because the text of textbox rendered by `input` tag is in the property and not in the HTML attribute, if we bind the attribute and change the value of the bound field, nothing occurs in rendered textfield.

```
<input value"@Name" id="name" />

@functions
{
    public string Name = "Monty";
}
 ```
If we use value instead of bind attribute, if `Name` field change, nothing is reflected in rendered textfield.

I propose a syntax to bind field to DOM property.

An possible example:
```
<input $value="@Name" />
```
This syntax, bind `Name` field to DOM property instead of HTML attribute.

Why?

- Consistency, instead of new attribute called `bind`, you know what is binding really.
- Some webcomponents, use new properties, for example:
```
// HTML
<my-calendar id="calendar" />

// JavaScript
var calendar = document.getElementById("calendar");
calendar.day; // 13
calendar.month; //4
calendar.year; // 2017
calendar.value; // undefined
```

In this case, I can't use `bind` attribute, but I can do:
```
<my-calendar $day="Day" $month="Month" $year="Year" />

@functions
{
    public int Day { get; set; }
    public int Month { get; set; }
    public int Year { get; set; }
}
```

**Comment by olavorn** [comment]
I agree that this is a versatille way to interact with DOM, but there are some aspects to consider:

1- DOM attribute changes may cause excessive repaints/redraws, should be used carefully.
2- Two way binding expects changes in attributes via javascript or other to apply to Blazor objects, in many cases other than value attribute and a tiny set of nodes, this may not occur, and it's a pathway to fundamental flaws cause it will be relying on browser events.
